skymap_scanner/server/start_scan.py

```diff
@@ -419,30 +466,54 @@ async def collect(
             if best.nside not in self.nsides_dict:
                 self.nsides_dict[best.nside] = {}
             if best.pixel in self.nsides_dict[best.nside]:
-                raise DuplicatePixelRecoException(
+                raise ExtraPixelRecoException(
                     f"NSide {best.nside} / Pixel {best.pixel} is already in nsides_dict"
                 )
             self.nsides_dict[best.nside][best.pixel] = best
             LOGGER.debug(f"Saved (found during {logging_id}): {best.id_tuple} {best}")
 
         # report after potential save
-        await self.reporter.record_pixreco(pixreco.nside, pixreco_start, pixreco_end)
+        await self.reporter.record_pixreco(
+            pixreco.nside,
+            pixreco_runtime,
+            roundtrip_start=self._pixreco_ids_sent__with_times[pixreco.id_tuple],
+            roundtrip_end=time.time(),
+        )
+
+    def is_scan_done(self) -> bool:
+        """Has every pixel been collected?"""
+        return self.n_received == self.n_sent
 
+    def ok_to_serve_more(self) -> bool:
+        """Return whether enough pixel-recos collected to serve more.
+
+        If we are approaching the end of the scan, always return False.
+        """
+        if self._end_game:
+            return False
```

Comment:
That's already implemented. The `Reporter` send a non-final ScanResult to SkyDriver (as opposed to saving to disk). From there, a SkyDriver user can build their plot, see https://github.com/icecube/skymap_scanner/blob/async-iterations/resources/utils/plot_skydriver_scan_result.py (this can move to https://github.com/icecube/[skyreader](https://github.com/icecube/skyreader), once it's set up)

Reply:
Ok great! In that case perhaps it'd be worth plotting a few intermediate results from a full iterative scan when testing this PR with SkyDriver, so we can see how the progression looks.